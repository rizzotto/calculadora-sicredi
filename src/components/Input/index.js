import React from "react";

import { TextField } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";

const useStyles = makeStyles({
    root: {
        width: "100%",

        "& .MuiOutlinedInput-root": {
            "& fieldset": {
                borderRadius: 8,
            },
        },
    },
});

export default function Input({ placeholder, onChange, value, ...rest }) {
    const classes = useStyles();

    return (
        <TextField
            classes={{ root: classes.root }}
            placeholder={placeholder}
            onChange={onChange}
            value={value}
            variant="outlined"
            {...rest}
        />
    );
}
Input.propTypes = {
    placeholder: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.string,
};
