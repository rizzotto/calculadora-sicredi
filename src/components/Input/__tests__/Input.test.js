import React from "react";
import renderer from "react-test-renderer";

import Input from "../index.js";

it("renders Input correctly", () => {
    const tree = renderer
        .create(
            <Input onChange={() => console.log("onChange")} value="onChange" />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
