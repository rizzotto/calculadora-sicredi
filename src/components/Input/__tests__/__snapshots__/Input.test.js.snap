// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="onChange"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-3"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
